### PR TITLE
Improve how the Jupyter notebooks are displayed in the documentation

### PR DESCRIPTION
**Pull request by seweber**
* Use [nbsphinx](https://nbsphinx.readthedocs.io/) to include notebooks in the documentation  (requires the installation of the Python packages *nbsphinx* and, perhaps, *ipython*).
* Add a gallery that shows previews of the notebooks.
* Add links to download the notebooks.
* To make it easier to write the documentation, a live preview of the documentation can be obtained via `make livehtml`.

### CODE REVIEW

**Review by hmenke — CHANGES_REQUESTED** [review]
As discussed earlier...

File: doc/sphinx/conf.py.in [hunk 1]
```diff
@@ -40,39 +39,59 @@ else:
...
+    "sphinx.ext.mathjax",
+    "sphinx.ext.viewcode",
+    "sphinx.ext.napoleon",
+    "sphinx.ext.extlinks",
+    "nbsphinx",
 ]
 
+# Disable cell numbers in jupyter notebooks and show link to ipynb file
+nbsphinx_prolog = """
+{% set docname = env.doc2path(env.docname, base=False).split("/")[-1] %}
```
Comment (hmenke) [inline doc/sphinx/conf.py.in:53]: This fixes the Ubuntu build error.
```suggestion
{% set docname = env.doc2path(env.docname, base=None).split("/")[-1] %}
```
Then we can revert to the old Docker image.

**Review by hmenke — CHANGES_REQUESTED** [review]

File: doc/sphinx/CMakeLists.txt [hunk 2]
```diff
@@ -19,57 +19,49 @@ find_program(SPHINX_BUILD_EXECUTABLE sphinx-build REQUIRED)
...
   "${SPHINX_BASE}/conf.py"
   @ONLY)
 
+# Directories to be copied to the build directory
+file(COPY "_static" DESTINATION "${SPHINX_BASE}")
+file(COPY "_templates" DESTINATION "${SPHINX_BASE}")
+
 # Prepare sphinx target
 add_custom_target(sphinx
-    ${CMAKE_COMMAND} -E env SPHINX_APIDOC_OPTIONS=members,inherited-members,show-inheritance
```
Comment (hmenke) [inline doc/sphinx/CMakeLists.txt:50]: Why was this removed?

Comment (hmenke) [reply doc/sphinx/CMakeLists.txt:50]: Once this has been adressed, please rebase and merge.

Comment (seweber) [reply doc/sphinx/CMakeLists.txt:50]: In my tests, calling sphinx-apidoc explicitly was not needed to generate the documentation for the pairinteraction module. Admittedly, I am not completely sure whether this has been just the case on my systems or whether it is in general not needed. I could move the options "members, inherited-members, ..." to conf.py.in so that these options are still applied. What do you think?

Comment (hmenke) [reply doc/sphinx/CMakeLists.txt:50]: The API documentation looks fine to me but the nbsphinx gallery is broken.

Comment (seweber) [reply doc/sphinx/CMakeLists.txt:50]: Once I encountered a similar issue with an old version of nbsphinx. I can try whether updating the version of nbsphinx in the CI solves the problem. If this works out, we could think about making a separate docker image with the newest nbsphinx version for building our documentation.

**Review by hmenke — CHANGES_REQUESTED** [review]
After rebase onto master this is ready to merge.

**Review by hmenke — APPROVED** [review]
